gomint-api/src/main/java/io/gomint/plugin/PluginManager.java

```diff
@@ -72,12 +75,33 @@
     /**
      * Register a new event listener for the given plugin. This only works when you call it from a plugin class.
      *
+     * The listener will be informaed of all events regardless of the world they take place in.
+     *
      * @param plugin   The plugin which wants to register this listener
      * @param listener The listener which we want to register
      * @throws SecurityException when somebody else except the plugin registers the listener
+     * @see #registerListener(Plugin, EventListener, ConcurrentHashMap.KeySetView)
+     * @since 2.0
      */
     PluginManager registerListener(Plugin plugin, EventListener listener);
 
+    /**
+     * Register a new event listener for the given plugin. This only works when you call it from a plugin class.
+     *
+     * Events implementing {@linkplain io.gomint.event.interfaces.WorldEvent WorldEvent} will be filtered for this
+     * listener, it will only be called for events taking place in worlds which {@linkplain World#folder() folder name}
+     * is present in the {@code worlds} collection.
+     *
+     * @param plugin   The plugin which wants to register this listener
+     * @param listener The listener which we want to register
+     * @param worlds   A whitelist of world {@linkplain World#folder() folder name}. Edits to the set are honored. Empty collection or {@code null}
+     *                 means all worlds.
+     * @throws SecurityException when somebody else except the plugin registers the listener
+     * @see #registerListener(Plugin, EventListener)
+     * @since 2.0
+     */
+    PluginManager registerListener(Plugin plugin, EventListener listener, @Nullable ConcurrentHashMap.KeySetView<String, Boolean> worlds);
```

Comment:
I don't like that a specific view is part of the api. Also this should be a Predicate system like i commented earlier in this review

Reply:
I thought you only wanted the internal EventManager to have the predicate system.

Reply:
No, the API should have that Predicate. The set should not CME, and if it does i would use a immutable set over a specific implementation.

Reply:
added